index.js

```diff
@@ -7,40 +7,66 @@ const app = express();
 const autovoice = require('./bin/lib/autovoice');
 const formatContentForReading = require('./bin/lib/formatContentForReading');
 
+const authS3O = require('s3o-middleware');
 
 var requestLogger = function(req, res, next) {
     debug("RECEIVED REQUEST:", req.method, req.url);
     next(); // Passing the request to the next handler in the stack.
 }
 
 app.use(requestLogger);
-app.use('/static', express.static('static'));
 
-app.get('/', (req, res) => {
-	res.sendFile(path.join(__dirname + '/static/index.html'));
-});
+// these routes do *not* have s3o
+
+app.use('/static', express.static('static'));
 
 app.get('/__gtg', (req, res) => {
 	res.status(200).end();
 });
 
+const PODCAST_TOKEN = process.env.PODCAST_TOKEN;
+if (! PODCAST_TOKEN ) {
+  throw new Error('ERROR: PODCAST_TOKEN not specified in env');
+}
+
 app.get('/podcast', (req, res) => {
   const rssUrl = req.query.rss;
-  const voice = req.query.voice;
-  autovoice.podcast(rssUrl, voice)
-  .then(feed => {
-    res.set('Content-Type', 'application/rss+xml');
-    res.send(feed);
-  })
+  const voice  = req.query.voice;
+  const token  = req.query.token;
+
+  if (! token)                        { res.status(400).send('This call requires a token parameter.'      ).end();
+  } else if (token !== PODCAST_TOKEN) { res.status(401).send('This call requires a valid token parameter.').end();
+  } else if(! voice)                  { res.status(400).send('This call requires a voice parameter.'      ).end();
+  } else if(! rssUrl)                 { res.status(400).send('This call requires a rss parameter.'        ).end();
+  } else {
+    autovoice.podcast(rssUrl, voice)
+    .then(feed => {
+      res.set('Content-Type', 'application/rss+xml');
+      res.send(feed);
+    })
+    ;
+  }
 });
 
 app.get('/audio.mp3', (req, res) => {
```

Comment:
Does this endpoint not also trigger a generation of audio content? Does it not require a token check too?